### ISSUE TITLE
Unable to run on Jenkins

### ISSUE THREAD

**Issue by ralphsmith80** [conventional-changelog/standard-version#474]
I've made lot's of comments on the [slack channel](https://app.slack.com/client/T09PUE50A/C6GK6LNSK).

TL;DR
I see the header information being updated in the `CHANGELOG.md` but no other information. There is other information that is should be present.

```
### 3.13.9-0 (2019-10-30)
### 3.13.8 (2019-10-28)
```

- Running the same code as Jenkins locally works although Jenkins fetches with `--no-tags` and that does produce different results, but still more content than just the header update.
- Running locally works as expected
- Running with `--dry-run` works as expected locally but fails on Jenkins

I know standard-version is proposed to be used locally as a stepping stone to semantic-release on a CI environment however I still need imperative versioning that standard-version supports and semantic-release does not.

**Comment by tommywo** [comment]
Hi @ralphsmith80,
Can you provide steps to reproduce this issue?

**Comment by ralphsmith80** [comment]
I can. It will require having a Jenkins instance that you can run against.

1. create a new npm package
1. add a pipeline script to run on Jenkins (script below)
1. push the new package to a repo
1. create a Jenkins multi-branch pipeline job that reference the repo and pipeline script
1. run the Jenkins job

Jenkins pipeline script:
```
pipeline {
  agent any
  tools {
    nodejs "node"
  }
  environment {
    CI = 'true'
  }
  parameters {
    choice(
      name: 'SEMVER_INCREMENT',
      choices: [
        'prerelease',
        'prepatch',
        'patch',
        'preminor',
        'minor',
        'premajor',
        'major'
      ],
      description: 'Specify the semver increment for `npm version`'
    )
  }
  stages {
    stage('Install') {
      steps {
        sh 'npm ci'
      }
    }
    stage('Version') {
      steps {
        sh "npm run release -- --release-as ${params.SEMVER_INCREMENT}"
        sh "cat CHANGELOG.md"
        sh "git log"
      }
    }
  }
}
```

### Dependencies
1. Jenkins will need to have the node plugin installed - note it's referenced in the tools section. You could also use docker but my Jenkins instance doesn't have docker. 

### Further considerations
Based on all the configurations I've done while debugging this I'm currently of the opinion that you could do any Jenkins configuration and it's not going to work. I added some log statements in the script about to ensure the `git log` and `CHANGELOG.md` shows what I expect. `git log` is showing what I expect but the `CHANGELOG.md` is not. 

I _think_ it might have to do with the way that Jenkins fetches the repo but I wrote a small script to mimic the Jenkins behavior and it work as expected locally. Are the specific requirement in how the repo must be fetched/cloned?

Test script to mimic Jenkins behavior:
```
git rev-parse --is-inside-work-tree # timeout=10
git config remote.origin.url ssh://git@stash.saas-p.com:7999/cb/cbe-ui.git # timeout=10
git --version # timeout=10
git config --get remote.origin.url # timeout=10
git fetch --tags --progress origin +refs/heads/*:refs/remotes/origin/* # timeout=10
git init ./bare-repo # timeout=10
git fetch --no-tags --progress ssh://git@stash.saas-p.com:7999/cb/cbe-ui.git +refs/heads/*:refs/remotes/origin/* # timeout=10
git config remote.origin.url ssh://git@stash.saas-p.com:7999/cb/cbe-ui.git # timeout=10
git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
git config remote.origin.url ssh://git@stash.saas-p.com:7999/cb/cbe-ui.git # timeout=10
git fetch --no-tags --progress ssh://git@stash.saas-p.com:7999/cb/cbe-ui.git +refs/heads/*:refs/remotes/origin/* # timeout=10
git config core.sparsecheckout # timeout=10
git checkout -f 526415ac8562f3c6e5c7d385210060bf0ecdfcc8 # timeout=10
git rev-list --no-walk 526415ac8562f3c6e5c7d385210060bf0ecdfcc8 # timeout=10

npm ci

npm run release -- --release-as prerelease
```

**Comment by ralphsmith80** [comment]
I did get this working but I had to use a newer instance of Jenkins and a docker agent but I have a new different issue now I'll open a separate issue for that.
